File: 2404-most-frequent-even-element/2404-most-frequent-even-element.java
class Solution {
    public int mostFrequentEven(int[] nums) {
        
        Arrays.sort(nums);
        int ans = -1;
        int maxi = Integer.MIN_VALUE;

        for(int i=0;i<nums.length;){
            if((nums[i] & 1) == 1){
                i++;
                continue;
            }
            int count = 0;
            int elem = nums[i];
            while(i<nums.length && nums[i] == elem){
                count++;
                i++;
            }

            if(count > maxi){
                maxi =  count;
                ans = elem;
            }
        }

        return ans;
    }
}

File: 2404-most-frequent-even-element/2404-most-frequent-even-element.cpp
class Solution {
public:
    int mostFrequentEven(vector<int>& nums) {
        unordered_map<int,int>mp;

        for(int i=0; i<nums.size(); i++){
            if( nums[i] %2 == 0){
               mp[nums[i]]++;
            }
        }
        if(mp.size()== 0) return -1;
        int maxi = INT_MIN;
        int even = 0;

        for(auto pr : mp){
            if(pr.second > maxi){
                maxi = pr.second;
                even = pr.first;
            }
            else if(pr.second == maxi){
                even = min(even,pr.first);
            }
        }

        return even;
       
    }
};